File: 5.implementation/NT/redsun/src/main/webapp/static/js/services/ProvidersService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('providerService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(provider) {
			var urlServer = $rootScope.makePostURL("/provider/insert");
			return $http.post(urlServer, provider);
		}
		
		// Update.
		, update: function(providerId, provider) {
			var urlServer = $rootScope.makePostURL("/provider/update/" + providerId);
			return $http.put(urlServer, provider);
		}
		
		// Delete.
		, delete: function(providerId) {
			var urlServer = $rootScope.makePostURL("/provider/delete/" + providerId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(providerId) {
			var url = $rootScope.makeGetURL("/provider/getbyid/" + providerId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, provider) {
			var url = $rootScope.makeGetURL("/provider/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: provider });
		}
		
	}// return.
});

File: 5.implementation/NXP/redsun-service/src/main/java/com/redsun/service/service/ProvidersService.java
package com.redsun.service.service;

import com.redsun.service.entities.Providers;
import com.redsun.service.entities.Result;

/**
 * Providers Service interface
 */
public interface ProvidersService {
	
    // Insert.
	Result insert(final Providers providers) ;

    // Update.
	Result update(final Providers providers) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listProvidersForPageAndFilter(final int itemsPerPage, final int pageNo, final Providers providers);

}
